### PR TITLE
Implementing Wallet Service

### PR DESCRIPTION
**Pull request by shayanmalinda**
This is a simple REST service to store the wallet address against user email in a MySQL database.

### CODE REVIEW

File: operations/wallet-service/utils.bal [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+    [jwt:Header, jwt:Payload] [_, payload] = check jwt:decode(jwt);
+    return check payload.cloneWithType();
+}
+
+# Check if string is empty or not.
+#
+# + val - String parameter
+# + return - If empty return TRUE, else return FALSE
+public isolated function isEmptyVal(string? val) returns boolean =>
+    val is () || val === "";
```
Comment (avishka95) [inline operations/wallet-service/utils.bal:24]: Shall we trim the value and check whether it's empty?
```suggestion
    val is () || val.trim().length() == 0;
```

File: operations/wallet-service/modules/database/db_functions.bal [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+# + return - True if wallet exists, false if not exists, error if error occurred
+public isolated function isUserWalletExists(string walletAddress) returns boolean|sql:Error {
+    types:UserWallet|sql:Error walletResponse = dbClient->queryRow(getUserWalletQuery(walletAddress));
+    if walletResponse is sql:NoRowsError {
+        return false;
+    } else if walletResponse is sql:Error {
+        log:printError("Error while checking user wallet", walletResponse, info = walletResponse.toString());
+        return walletResponse;
+    }
+    return true;
```
Comment (avishka95) [inline operations/wallet-service/modules/database/db_functions.bal:24]:
```suggestion
    if walletResponse is sql:Error && walletResponse !is sql:NoRowsError {
        log:printError("Error while checking user wallet", walletResponse, info = walletResponse.toString());
        return walletResponse;
    }
    
    return walletResponse is types:UserWallet;
```
I feel that the above looks better. Test this before implementing.

Comment (shayanmalinda) [reply operations/wallet-service/modules/database/db_functions.bal]: [8c5ce54](https://github.com/wso2/digiops-currency/pull/22/commits/8c5ce54e2093f7e170daf87793924ef8b54612a4)

File: operations/wallet-service/utils.bal (hunk 1 above)
Comment (shayanmalinda) [reply operations/wallet-service/utils.bal]: [c4387a0](https://github.com/wso2/digiops-currency/pull/22/commits/c4387a0436f24794e9f952117b96b8b4e558b5cb)

File: operations/wallet-service/modules/database/clients.bal [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+//
+// This software is the property of WSO2 LLC. and its suppliers, if any.
+// Dissemination of any information or reproduction of any material contained
+// herein in any form is strictly forbidden, unless permitted by WSO2 expressly.
+// You may not alter or remove any copyright or other notice from copies of this content.
+
+import ballerinax/mysql;
+import ballerinax/mysql.driver as _;
+
+configurable Database connectorStoreDatabase = ?;
```
[the remark this answers is not in the document]
Comment (shayanmalinda) [reply operations/wallet-service/modules/database/clients.bal]: My Bad, Thanks Yukthi
[d1dd543](https://github.com/wso2/digiops-currency/pull/22/commits/d1dd5434fdff407800c52da5955bc83d500a57a9)

**Review by yuk7hi — APPROVED** [review]

**Pull request merged**